crates/driver/src/infra/solver/dto/notification.rs

```diff
@@ -64,12 +70,23 @@ pub struct Notification {
 #[serde(rename_all = "lowercase")]
 pub enum Kind {
     EmptySolution,
+    SimulationFailed(Tx),
     ScoringFailed(ScoreKind),
     NonBufferableTokensUsed(BTreeSet<H160>),
     SolverAccountInsufficientBalance(U256),
     DuplicatedSolutionId,
 }
 
+#[serde_as]
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Tx {
```

Comment:
The reason we have this struct is to adhere the DDD that wants to separate the domain object and dto object (where the serialization is encapsulated).
The reason I dropped access list is that access lists don't influence simulation status (success/revert) therefore are not needed to be part of this struct that is mainly used to reproduce the simulation.

Reply:
> The reason we have this struct is to adhere the DDD that wants to separate the domain object and dto object (where the serialization is encapsulated).

I see, it's about separating core logic and interface, if I understood DDD design principle. All good then.

> access lists don't influence simulation status (success/revert)

This isn't the case as far as I know, for example sending ETH to a safe could fail if the access list doesn't account for it.

Reply:
I tried to find if this is still valid, since we already talked about dropping access lists completely. 
However, in the meantime, I added access list as you suggested. I used the `web3::types::AccessList` as I assume this one is the most convenient since all the data is already envisioned to be used to recreate tx for simulation.